testing/python/approx.py

```diff
@@ -496,3 +496,13 @@ def test_numpy_scalar_with_array(self):
         assert actual != approx(expected, rel=5e-8, abs=0)
         assert approx(expected, rel=5e-7, abs=0) == actual
         assert approx(expected, rel=5e-8, abs=0) != actual
+
+    def test_generic_iterable_sized_object(self):
+        class newIterable(object):
+            def __iter__(self):
+                return iter([1, 2, 3, 4])
+
+            def __len__(self):
+                return 4
+
+        assert [1, 2, 3, 4] == approx(newIterable())
```

Comment:
i believe his breaks if an approx object is reused - just unpack explicitly

Reply:
Not entirely sure what you meant - unless you read the test class as an iterator rather than an iterable? In any case, I've split out the creations, was this what you meant?

https://github.com/pytest-dev/pytest/pull/4327/files#diff-a4d4d6cc8ecc32438c8a4da391852f22R508